sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -844,6 +879,22 @@ public void processElement(ProcessContext c, BoundedWindow window) throws Except
                   failures.add(new BigtableWriteException(c.element(), exception));
                 }
               });
+      if (config.getBulkMutationThrottling()) {
+        long delta = 0;
```

Comment:
you can probably get the instance outside of the lock: 

ResourceLimiterStats stats = ResourcesLimiterStats.getInstance(..);

and inside the lock:
long newAgg = stats.getCumulativeThrottlingTimeNanos();
delta = ...